### PR TITLE
RavenDB-18998 - support for sharded skip and take

### PR DESCRIPTION
**Pull request by grisha-kotler**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18998/Sharding-Queries-Handle-skip-and-take

### Additional description

Sharding - Queries - Handle skip and take.
This is only to support v4.x and v5.x clients.
For the v6.x client we'll have a `ContinuationToken`.

### How risky is the change?

- Low 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### CODE REVIEW

**Comment by ppekrol** [comment]
Rebase and check failing tests

File: src/Raven.Server/Documents/Sharding/Queries/ShardedQueryProcessor.cs [hunk 1]
```diff
@@ -165,8 +165,12 @@ private void RewriteQueryForPaging(ref BlittableJsonReaderObject queryTemplate)
         var limit = ((_query.Limit ?? 0) + (_query.Offset ?? 0)) * (long)_parent.DatabaseContext.ShardCount;
         if (limit > int.MaxValue) // overflow
             limit = int.MaxValue;
+
         modifiedArgs[limitToken] = limit;
 
+        queryTemplate.Modifications.Remove(nameof(IndexQueryServerSide.Start));
+        queryTemplate.Modifications.Remove(nameof(IndexQueryServerSide.PageSize));
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Sharding/Queries/ShardedQueryProcessor.cs:172]: Those can be sent by older clients, right?

**Comment by arekpalinski** [comment]
I'm merging this to avoid hard conflicts. Please check my questions in the above comment

File: src/Raven.Server/Documents/Sharding/Queries/ShardedQueryProcessor.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Server/Documents/Sharding/Queries/ShardedQueryProcessor.cs:172]: It is sent by older clients and is translated correctly into RQL.
Than `Start` and `PageSize` are used in `IndexQueryServerSide`.
When we send it again to the shards we need to remove it.

**Pull request merged**